### PR TITLE
EWPP-1049: Address field should have only Country as mandatory.

### PR DESCRIPTION
**Pull request by yenyasinn**

### CODE REVIEW

File: modules/oe_theme_helper/src/Plugin/Field/FieldFormatter/AddressInlineFormatter.php [hunk 1]
```diff
@@ -100,6 +100,7 @@ protected function viewElement(AddressInterface $address, $langcode) {
     else {
       $format_string = $address_format->getFormat() . "\n" . '%country';
     }
+    $format_string = str_replace([',', ' - '], "\n", $format_string);
```
Comment (yenyasinn) [inline modules/oe_theme_helper/src/Plugin/Field/FieldFormatter/AddressInlineFormatter.php:103]: It is fix for https://citnet.tech.ec.europa.eu/CITnet/jira/browse/EWPP-1050

File: modules/oe_theme_helper/tests/src/Kernel/Plugin/Field/FieldFormatter/AddressInlineFormatterTest.php [hunk 2]
```diff
@@ -29,20 +29,96 @@ protected function setUp(): void {
 
   /**
    * Tests formatting of address.
+   *
+   * @param array $address
+   *   Address in the field format.
+   * @param string $expected
+   *   Expected result.
+   *
+   * @dataProvider addressFieldDataProvider
```
Comment (upchuk) [inline modules/oe_theme_helper/tests/src/Kernel/Plugin/Field/FieldFormatter/AddressInlineFormatterTest.php]: Yeah, let's not do a dataprovider here. The theme tests take a very long time as it is. Can you just loop through an array of test data and create the entity, render, assert and delete? I don't think you even need to save to render.

**Review by upchuk — CHANGES_REQUESTED** [review]

File: modules/oe_theme_content_event/src/Plugin/ExtraField/Display/EventExtraFieldBase.php [hunk 3]
```diff
@@ -53,4 +54,25 @@ public static function create(ContainerInterface $container, array $configuratio
...
+  public function getVenueInlineAddress(VenueInterface $venue): string {
+    $location = [];
+    $renderable = $this->entityTypeManager->getViewBuilder('oe_venue')->viewField($venue->get('oe_address'));
+    foreach (['locality', 'country'] as $field) {
+      if (!empty($renderable[0][$field]['#value'])) {
+        $location[] = $renderable[0][$field]['#value'];
+      }
+    }
+
+    return implode(', ', $location);
```
Comment (upchuk) [inline modules/oe_theme_content_event/src/Plugin/ExtraField/Display/EventExtraFieldBase.php]: I know it was like this before but like this we may lose cacheability, so let's render the field before, like we do here: https://github.com/ec-europa/ewcms/pull/1015/files#diff-0a7bee81a5ae430d8258164da937320de3cf918fd4c77ffaff231215d8294f26R463

**Review by upchuk — CHANGES_REQUESTED** [review]

File: modules/oe_theme_content_event/src/Plugin/ExtraField/Display/EventExtraFieldBase.php (hunk 3 above)
Comment (yenyasinn) [reply modules/oe_theme_content_event/src/Plugin/ExtraField/Display/EventExtraFieldBase.php]: Done, but refactoring of Event's extra fields were required for it.

File: modules/oe_theme_helper/tests/src/Kernel/Plugin/Field/FieldFormatter/AddressInlineFormatterTest.php (hunk 2 above)
Comment (yenyasinn) [reply modules/oe_theme_helper/tests/src/Kernel/Plugin/Field/FieldFormatter/AddressInlineFormatterTest.php]: Done

File: modules/oe_theme_content_event/src/Plugin/ExtraField/Display/DetailsExtraField.php [hunk 4]
```diff
@@ -152,41 +142,69 @@ protected function getRenderableDates(ContentEntityInterface $entity): array {
...
-      return;
-    }
-
-    $venue = $entity->get('oe_event_venue')->entity;
-    if (!$venue instanceof VenueInterface) {
-      return;
+    if (!$entity->get('oe_event_venue')->isEmpty() && $entity->get('oe_event_venue')->entity instanceof VenueInterface) {
+      $address = $this->getVenueInlineAddress($entity->get('oe_event_venue')->entity);
+      if (!empty($address)) {
+        // Only display locality and country, inline.
```
Comment (drishu) [inline modules/oe_theme_content_event/src/Plugin/ExtraField/Display/DetailsExtraField.php]: its clear from below code, no need

File: modules/oe_theme_content_event/src/Plugin/ExtraField/Display/DetailsExtraField.php [hunk 5]
```diff
@@ -152,41 +142,69 @@ protected function getRenderableDates(ContentEntityInterface $entity): array {
...
+
+  /**
+   * Add event online type to event details.
+   *
+   * @param array $build
+   *   Render array.
+   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+   *   Content entity.
+   */
+  protected function addRenderableOnlineType(array &$build, ContentEntityInterface $entity) {
```
Comment (drishu) [inline modules/oe_theme_content_event/src/Plugin/ExtraField/Display/DetailsExtraField.php]: just for good practice
```suggestion
  protected function addRenderableOnlineType(array &$build, ContentEntityInterface $entity): void {
```

**Review by drishu — CHANGES_REQUESTED** [review]

**Review by drishu — DISMISSED** [review]

File: modules/oe_theme_content_event/src/Plugin/ExtraField/Display/DetailsExtraField.php [hunk 6]
```diff
@@ -46,10 +38,12 @@ class DetailsExtraField extends EventExtraFieldBase implements ContainerFactoryP
...
    *   The entity repository service.
+   * @param \Drupal\Core\Render\RendererInterface $renderer
+   *   The renderer service.
    */
-  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, EntityRepositoryInterface $entity_repository) {
-    parent::__construct($configuration, $plugin_id, $plugin_definition, $entity_type_manager);
-    $this->entityRepository = $entity_repository;
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, EntityRepositoryInterface $entity_repository, RendererInterface $renderer) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $entity_type_manager, $entity_repository);
+    $this->renderer = $renderer;
```
Comment (sergepavle) [inline modules/oe_theme_content_event/src/Plugin/ExtraField/Display/DetailsExtraField.php:65]: I believe this property should be declared within class.

**Review by sergepavle — DISMISSED** [review]

**Review by sergepavle — DISMISSED** [review]

**Review by 22Alexandra — DISMISSED** [review]

File: modules/oe_theme_content_event/src/Plugin/ExtraField/Display/DetailsExtraField.php [hunk 7]
```diff
@@ -46,10 +52,13 @@ class DetailsExtraField extends EventExtraFieldBase implements ContainerFactoryP
    *   The entity type manager.
    * @param \Drupal\Core\Entity\EntityRepositoryInterface $entity_repository
    *   The entity repository service.
+   * @param \Drupal\Core\Render\RendererInterface $renderer
+   *   The renderer service.
    */
-  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, EntityRepositoryInterface $entity_repository) {
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, EntityRepositoryInterface $entity_repository, RendererInterface $renderer) {
```
Comment (upchuk) [inline modules/oe_theme_content_event/src/Plugin/ExtraField/Display/DetailsExtraField.php]: Like I said, you need to default the new argument to NULL and check if it doesnt not implement the correct interface, load it statically.

**Review by upchuk — CHANGES_REQUESTED** [review]

File: modules/oe_theme_content_event/src/Plugin/ExtraField/Display/DetailsExtraField.php (hunk 7 above)
Comment (yenyasinn) [reply modules/oe_theme_content_event/src/Plugin/ExtraField/Display/DetailsExtraField.php]: Fixed.

**Review by imanoleguskiza — DISMISSED** [review]

**Review by upchuk — APPROVED** [review]

**Review by imanoleguskiza — APPROVED** [review]

**Pull request merged**